airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/commands.py

```diff
@@ -99,14 +99,43 @@ def validate_environment(is_local: bool, use_remote_secrets: bool):
             "GCP_GSM_CREDENTIALS",
             "CI_REPORT_BUCKET_NAME",
             "CI_GITHUB_ACCESS_TOKEN",
+            "DOCKER_HUB_USERNAME",
+            "DOCKER_HUB_PASSWORD",
         ]
         for required_env_var in required_env_vars_for_ci:
             if os.getenv(required_env_var) is None:
                 raise click.UsageError(f"When running in a CI context a {required_env_var} environment variable must be set.")
-    if use_remote_secrets and os.getenv("GCP_GSM_CREDENTIALS") is None:
-        raise click.UsageError(
-            "You have to set the GCP_GSM_CREDENTIALS if you want to download secrets from GSM. Set the --use-remote-secrets option to false otherwise."
-        )
+
+
+def should_use_remote_secrets(use_remote_secrets: Optional[bool]) -> bool:
+    """Check if the connector secrets should be loaded from Airbyte GSM or from the local secrets directory.
+
+    Args:
+        use_remote_secrets (Optional[bool]): Whether to use remote connector secrets or local connector secrets according to user inputs.
+
+    Raises:
+        click.UsageError: If the --use-remote-secrets flag was provided but no GCP_GSM_CREDENTIALS environment variable was found.
+
+    Returns:
+        bool: Whether to use remote connector secrets or local connector secrets.
+    """
+    gcp_gsm_credentials_is_set = bool(os.getenv("GCP_GSM_CREDENTIALS"))
+    if use_remote_secrets is None:
+        if gcp_gsm_credentials_is_set:
+            main_logger.info("GCP_GSM_CREDENTIALS environment variable found, using remote connector secrets.")
+            return True
+        else:
+            main_logger.warning("No GCP_GSM_CREDENTIALS environment variable found, using local connector secrets.")
```

Comment:
```suggestion
            main_logger.info("No GCP_GSM_CREDENTIALS environment variable found, using local connector secrets.")
```
I'd say this is an info (since no one set it explicitly), and the behavior of if the secrets folder is then empty is a warning